### ISSUE TITLE
Incorrect popover placement [6.2]

### ISSUE THREAD

**Issue by ndiego** [WordPress/gutenberg#48140]
### Description

First reported in Trac: [#57730](https://core.trac.wordpress.org/ticket/57730)

In 6.2 Beta 2 without Gutenberg active, the position of the block inserter popover is occasionally incorrect. See the gifs below. Note this might be related to https://github.com/WordPress/gutenberg/issues/47923.

### Step-by-step reproduction instructions

1. Using 6.2 Beta 2, create a new post/page.
2. Using the inserter in the Editor, add an image block. Notice that the popover is in the correct position.
3. Again, using the inserter in the Editor, go to add another block. Notice that the popover is located too far below the inserter button. The results were even more dramatic in [#57730](https://core.trac.wordpress.org/ticket/57730). 

### Screenshots, screen recording, code snippet

**WordPress 6.2 Beta 2 (Incorrect position)**

![popover-6-2](https://user-images.githubusercontent.com/4832319/219459701-00f9dbae-76c9-4d44-a781-90f595e917fb.gif)

**WordPress 6.2 Beta 2 + Gutenberg Trunk (No Issues)**

![popover-6-2-gutenberg](https://user-images.githubusercontent.com/4832319/219459744-5f37bec4-5b33-4cb3-ab44-2b421f121077.gif)

**WordPress 6.1.1 (No Issues)**

![popover-6-1](https://user-images.githubusercontent.com/4832319/219459688-fa3209cc-a3f4-47e5-b3a5-051c0c9f8f1a.gif)

### Environment info

- WordPress 6.2 Beta 2

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
I was also able to reproduce it in WordPress 6.2 Beta2.

If you make any changes to the content, perhaps as shown below, the popover will be misaligned when you click on the root level inserter:

https://user-images.githubusercontent.com/54422211/219910060-8946ad2c-ee28-417e-8262-69a63a9c044e.mp4

---

Curiously, this problem is not reproduced in the classic theme:

https://user-images.githubusercontent.com/54422211/219910068-b65fb167-6506-40ce-91ab-c8804e0d79d2.mp4

**Comment by paaljoachim** [comment]
I happen to open a new issue which I have now closed as a duplicate.
https://github.com/WordPress/gutenberg/issues/48234
Here is the initial comment I made.

Using the Site Editor.
Open List View. Then explore various blocks such as Nav block. Add a new link. Notice the modal is partly outside the right edge of the browser.
Then add a new link in the Social Links block. Again notice the modal is even further outside the right edge of the browser.

**Step-by-step reproduction instructions**
Go to the Site Editor.
Open List View.
In the Nav block add a new link. Notice location of popup link modal.
Add a Social Links block and add a service.
Notice the location of the modal popup.
Screenshots, screen recording, code snippet
Here is a reply:
https://app.replay.io/recording/page-template-editor-bluewinddesign-wordpress--1c3580d8-3826-4de8-ad21-5bf233de809d

**Environment info**
WordPress 6.2 beta 2
WordPress Beta Tester plugin
Twenty Twenty Three
